src/v/cluster/rm_stm.cc

```diff
@@ -692,6 +692,23 @@ ss::future<tx_errc> rm_stm::do_commit_tx(
 
 abort_origin rm_stm::get_abort_origin(
   const model::producer_identity& pid, model::tx_seq tx_seq) const {
+    if (is_transaction_ga()) {
+        auto tx_seqs_it = _log_state.tx_seqs.find(pid);
```

Comment:
Put the `_log_state.tx_seq` and `_mem_state.tx_seq` checks after `prepared_it` for `!is_transaction_ga()` case too. `old'` should support `old` and `new`!